src/services/services.ts

```diff
@@ -3961,18 +3975,15 @@ module ts {
             if (results) {
                 let sourceFile = getCanonicalFileName(normalizeSlashes(fileName));
 
-                // ensure the results are in the file we're interested in
-                results.forEach((value) => {
-                    let targetFile = getCanonicalFileName(normalizeSlashes(value.fileName));
-                    Debug.assert(sourceFile == targetFile, `Unexpected file in results. Found results in ${targetFile} expected only results in ${sourceFile}.`);
-                });
+                // Get occurrences only supports reporting occurrences for the file queried.  So 
+                // filter down to that list.
+                results = filter(results, r => r.fileName === fileName);
             }
 
             return results;
         }
 
-        /// References and Occurrences
-        function getOccurrencesAtPositionCore(fileName: string, position: number): ReferenceEntry[] {
+        function getDocumentHighlights(fileName: string, position: number): DocumentHighlights[] {

```

Comment:
Ugh, the diff is awful.  All that happened was that the code to get lexical occurrences was moved into a function called getSyntacticDocumentHighlights.  This returned the new DocumentHighlights type.  getSemanticDocumentHighlights now converts the results of getReferencedSymbols.  And the now deprecated getOccurrencesAtPosition function just defers to getDocumentHighlights and converts the results to its older API.

One of the new benefits of the getDocumentHighlights api is that it does not need to limit results just to the file it is in.  For example, roslyn will show highlights across all open files.